controllers/changetierrequest/changetierrequest_controller.go

```diff
@@ -208,6 +226,29 @@ func (r *Reconciler) changeTier(logger logr.Logger, changeTierRequest *toolchain
 	return nil
 }
 
+func (r *Reconciler) changeTierInSpace(logger logr.Logger, changeTierRequest *toolchainv1alpha1.ChangeTierRequest, namespace string) error {
+	space := &toolchainv1alpha1.Space{}
+	if err := r.Client.Get(context.TODO(), types.NamespacedName{
+		Namespace: namespace,
+		Name:      changeTierRequest.Spec.MurName,
+	}, space); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("No Space found for ChangeTierRequest")
+			return nil
+		}
+		return r.wrapErrorWithStatusUpdate(logger, changeTierRequest, r.setStatusChangeFailed, err, "unable to get Space with name %s", changeTierRequest.Spec.MurName)
+	}
+	space.Spec.TierName = changeTierRequest.Spec.TierName
+	// remove the TemplateTierHash label on the Space resource (and let the SpaceController set it to the latest value)
+	delete(space.Labels, tierutil.TemplateTierHashLabelKey(space.Spec.TierName))
```

Comment:
Well, this is changed in one update call, so both of the values are updated together - it should not happen that the tierName is changed but the label stays there. Correct me if I'm wrong.
A different story is a case when the `ChangeTierRequest` doesn't change the `space.Spec.TierName` at all:
1. `Space.Spec.TierName=advanced`
2. `ChangeTierRequest.Spec.TierName=advanced` 
3. Then the `ChangeTierRequest` controller only deletes the label but doesn't change the spec. However, the `Space` controller won't be triggered since it doesn't watch the changes of the labels, so it won't set the label back.
There should be something like:
```go
if space.Spec.TierName == changeTierRequest.Spec.TierName {
	return nil
}
space.Spec.TierName = changeTierRequest.Spec.TierName
// remove the TemplateTierHash label on the Space resource (and let the SpaceController set it to the latest value)
delete(space.Labels, tierutil.TemplateTierHashLabelKey(space.Spec.TierName))
```

Reply:
done in 2f3805e